---
title: "Proteome-wide Screen for RNA-dependent Proteins in HeLa Cells Synchronized in Interphase"
subtitle: Data Analysis Project 2023
output:
  pdf_document: default
  word_document: default
mainfont: Inter
header-includes:
  - \usepackage[default]{inter}
  - \usepackage[T1]{fontenc}
  - \usepackage[default]{inter}
  - \usepackage[T1]{fontenc}
  - \usepackage{titling}
  - \pretitle{\begin{center}\huge\bfseries}
  - \posttitle{\end{center}} 
geometry: "left=1.5cm,right=1.5cm,top=1.5cm,bottom=2cm"
fontsize: 11pt
---
\pagenumbering{gobble} 
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, dev="cairo_pdf")
load("Data/Environment.RData")
load("Data/Table_HS_Non_RBP.RData")
```

\begin{center}
Hannah Brehm, Johann Blakytny, Kira Hoffmann, Viktor Bonin  

Supervisor: Dr. Maïwen Caudron-Herger  

Tutor: Fabio Rauscher  

17.07.2023
\end{center}

<br>  
<br>  
<br>  
<br>  


***
# Abstract {.unlisted .unnumbered}
RNA-protein interactions are key mediators of integral cellular and metabolic processes. Therefore, alterations in RNA-binding proteins play a central role in the initiation and progression of many human diseases. The method R-DeeP allows to investigate the RNA-dependence of proteins, meaning that not only RNA-protein interactions are detected but also RNA-protein-protein interactions. Untreated and RNase-treated cell lysates are separated on a sucrose gradient into fractions. The proteins in these fractions are analyzed through mass spectrometry. Based on this spectrometry data, we aim to identify RNA-dependent proteins (RDPs). After cleaning the data, the protein distributions were smoothed using a three-fraction window. Global and local maxima of the smoothed protein distributions were then determined. Each global maximum and the highest local maximum were fitted with Gaussian curves. The subsequent data was used to define a set of selection criteria. The optimal selection criteria for identifying RDPs were found using principal component analysis combined with manually determined shifting behavior. For each optimal selection criterion, a threshold for RNA-dependence was set. This resulted in the identification of 539 RDPs, 274 partial RDPs, and 5177 non-RDPs. These findings were compared to the RBP2GO database. Linear regression was performed to investigate if the mass spectrometry data could be used to examine cellular parameters, such as the kinetics of RNA-protein interactions.  
This analysis shows how important bioinformatical methods are in the identification of RDPs. Future work based on R-DeeP and similar methods will undoubtedly have important implications in the treatment of many human diseases. 
<br>  
<br>  

***
\newpage

\renewcommand*\contentsname{Table of Contents}
\setcounter{tocdepth}{2}
\tableofcontents

\newpage
\pagenumbering{arabic} 
# 1. Introduction
RNA and its interacting proteins are relevant for essential cellular activity. Especially in posttranscriptional processes, RNA-binding proteins (RBPs) mediate the regulation of posttranscriptional modification, transport through the cytoplasm, translation, and stability of the RNA template. Different RNA-binding motifs have been characterized so far allowing the linkage of proteins to ribonucleic acid (Burd and Dreyfuss, 1994). The structure of most RNA-binding domains is highly conserved; nevertheless, many RBPs bind with high specificity and affinity to distinct RNAs (Lunde *et al.*, 2007). As RBPs are crucial for certain housekeeping mechanisms, mutations or other alterations in human RBPs can have pathological effects; for instance, alterations in the expression of the *zinc-finger protein SNAI1* are associated with cancer metastasis (Gebauer *et al.*, 2021).

There are multiple classifications for interactions between proteins and RNA. RBPs can be characterized in a binary system of being “specific” or “nonspecific” where specific RBPs interact with defined RNA motifs and nonspecific RBPs bind to RNA with an absence of definite motifs (Jankowsky and Harris, 2015). Further differentiation can lead to RBP classes defined by specific RNA-binding domains such as zinc-finger domains or RNA recognition motifs. Additionally, the targets of RBPs can be considered for creating subgroups. Thus, for instance mRNA‑binding or tRNA‑binding RBPs are classified (Gerstberger *et al.*, 2014).

Furthermore, RBPs can be classified as a subcategory of RNA-interacting proteins themselves. The concept of RNA-dependence combines RBPs with proteins which are interacting indirectly with RNA. The latter can also be called RBP-interacting proteins. Therefore, RNA-dependent proteins (RDPs) allow for a broader comprehension of the RNA interactome (Caudron-Herger *et al.*, 2020b).

Various methods have been applied to identify RNA recognition elements so far, for example crosslinking and immunoprecipitation followed by sequencing (CLIP-seq) (Gerstberger *et al.*, 2014; 2013). This enables identification of RNAs bound to proteins. Detecting the proteins interacting with RNA requires other methods such as comprehensive identification of RBPs by mass spectrometry (CHIRP-MS) (Licatalosi *et al.*, 2020). Introducing the method R-DeeP, Caudron-Herger *et al.* promise a “proteome-wide, unbiased, and enrichment-free screen” that is solely based on disrupting the RNA-protein and RNA-protein-protein interaction (Caudron-Herger *et al.*, 2020b; 2019). Through R-DeeP, we aim to identify more human RBPs and RBP-interacting proteins in this project.

# 2. Material and Methods

## 2.1 Referenced Data
The protocol for an R-DeeP Screen was performed on HeLa cells synchronized in interphase (Caudron-Herger *et al.*, 2020b): Untreated and RNase-treated cell lysates were separated into 25 fractions on a sucrose gradient. The proteins in these fractions were identified and their amount quantified using mass spectrometry. This yielded the distribution of each protein over the 25 fractions with three replicates per fraction.   

## 2.2 Data Cleanup
The original mass spectrometry dataset was split into two separate datasets containing the values for all RNase fractions and Control fractions, respectively. The following steps were performed on both datasets.

### 2.2.1 NA Values and Zero Rows
All rows containing only zeros, of which there were five, were deleted from the datasets. No NA values were found.

### 2.2.2 Pearson Correlation
The Pearson correlation was used to test the reproducibility of the data. In seven cases, one of the three replicates contained only zeros. These replicates were replaced with NA values as determining the Pearson correlation with these values would have been impossible. In five rows, two out of the three replicates contained only zeros. These genes were removed completely from the datasets.

The Pearson correlation $r$ was computed as follows for each pair of replicates at a time using the sample size $N$, individual sample points $x_i$ and $y_i$, sample means $\bar{x}$ and $\bar{y}$ and standard deviations $s_x$ and $s_y$: \[r = \frac{1}{N-1} \sum_{i = 1}^{N}{\frac{x_i - \bar{x}}{s_x} \frac{y_i - \bar{y}}{s_y}}\]

For each gene and condition this resulted in three correlation coefficients. The genes for which all three correlation coefficients fell below a threshold of 0.95 were removed from both datasets. This applied to 1086 genes. If two out of three correlation coefficients were under 0.95 the replicate associated with both coefficients was replaced with NA. Lastly, if one of three correlation coefficients was below the determined threshold the replicate with the lower value in the other associated correlation coefficient was replaced with NA.  

### 2.2.3 Mean and Standard Error of Replicates
After testing for reproducibility and deleting unreliable values, the arithmetic mean and standard error were calculated for all triplicates and used for further analysis.


## 2.3 Smoothing and Determination of Maxima

### 2.3.1 Smoothing the Data
To determine the global and local maxima of the datasets, the Control and RNase datasets were smoothed separately. Otherwise, the determination would have been influenced by outliers and wrong peaks.  

Smoothing was achieved by calculating the mean protein value for three neighboring fractions for each fraction with the following formula (applies to all fractions between 2 and 24): \[y = \frac{f(x-1)+f(x)+f(x-1)}{3}\]    
*$x$: fraction position ($2  \leq  x  \leq  24$);* $\quad$ *$f(x)$: mass spectrometry value at fraction position $x$;* $\quad$ *$y$: smoothed value for fraction position $x$*  
  
For fraction positions 1 and 25, the mean of fractions 1 and 2 and accordingly 24 and 25 were used instead.

### 2.3.2 Normalization of Smoothed Data
In order to be able to compare the different protein amounts, the smoothed datasets were normalized to 100. This means the sum of the mass spectrometry data in the different fractions for one protein added up to 100: \[y = \frac{f(x)}{\sum_{x=1}^{25}f(x)} *100\]     
*$x$: fraction position ($1 \leq x \leq 25$);* *$f(x)$: mass spectrometry value at fraction position $x$;* *$y$: normalized value at fraction position $x$*  
The same procedure was then applied to the standard error.

### 2.3.3 Determination of Maxima
If a protein is RNA-dependent, the maximum amount of protein should be located in a different fraction in the RNase sample compared to the Control sample. To determine for which proteins this is the case, the global and local maxima of the mass spectrometry data were determined.  
  
### 2.3.3.1 Global Maxima
The global maxima of each protein for the smoothed and normalized datasets were determined by identifying the fraction which contained the highest mass spectrometry value. For proteins that had two or three fractions with the highest value, the mean fraction position was used as the fraction position of the global maximum.  
  
### 2.3.3.2 Local Maxima
The local maxima for the smoothed and normalized datasets were determined by comparing the protein amount of each fraction to the protein amounts in the four neighboring fractions. If the protein amount in that fraction is greater or equal to its neighborhood, it was defined as a local maximum. For the fraction positions 2 and 24, the protein amounts had to be greater than or equal to the protein amounts in the fractions 1, 3, and 4 or fractions 22, 23, and 25, respectively. The fraction positions 1 and 25 had to be greater than or equal to fractions 2 and 3 and fractions 23 and 24, respectively. For proteins that had local maxima consisting of two or three neighboring fractions with the same mass spectrometry value, the mean fraction position was calculated. To distinguish local from global maxima, all global maxima were subtracted from the local maxima dataset.  

## 2.4 Gaussian Fitting
A Gaussian curve was fitted to the global peak and the local peak with the highest protein amount of each gene in the Control and RNase samples. A Gaussian curve $f(x)$ can be described using the height $h$, standard deviation $\sigma$ and mean $\mu$ as: \[f(x) = \frac {h} {\sqrt{2\pi} \cdot \sigma } \cdot e^{-\frac {1} {2} (\frac {x - \mu} {\sigma})^{2}}\]
These parameters were first estimated and then optimized. $\mu$ was estimated as the fraction position of the global peak and highest local peak, $\sigma$ and $h$ were estimated using the half width at half maximum (HWHM) and maximum protein amount (MPA): \[\sigma =\frac{HWHM}{\sqrt{2 \cdot ln(2)}};\qquad h = {MPA} \cdot \sqrt{2\pi} \cdot \sigma\] 
These estimated values were optimized by minimizing the sum of squares of the differences between the protein amount values of the Gaussian curve $G$ and the smoothed and normalized protein distribution $p$ ($min(\sum(G-p)^2)$). The optimized parameters were used to calculate the protein amount under the Gaussian curve and the overlap of the Gaussian curves between the Control and the RNase samples. 

## 2.5 Selection Criteria for the Identification of RNA-Dependent Proteins
To determine the optimal selection criteria for an RNA-dependent protein, multiple selection criteria were compiled and applied to a principle component analysis. The best criteria were subsequently used to differentiate between RDPs, partial RDPs, and non-RDPs. 

### 2.5.1 Definition of Selection Criteria
The following criteria were compiled to determine the optimal selection criteria for defining RNA-dependent proteins:

The **absolute peak shift** 
between the Control and RNase global peaks, 
between the global and main local peak of Control,
and between the global and main local peak of RNase,   

the **absolute change** 
in the **standard deviation** between Control and RNase, 
in the **global peak height** between Control and RNase, 
and between the the **protein amount** under the Gaussian curves of the Control and RNase peaks,  

the **percentage of protein amount**
that is lost under the Control global peak after RNase treatment,
and that is gained under the RNase global peak after RNase treatment,  

the **absolute differences in the protein amounts**
of the Control and RNase samples under the main local Control peak,  

the **percentage of overlap**
between the Gaussian curves of the Control and RNase,
between the Gaussian curves of the main local peak and global peak of Control,
between the Gaussian curves of the main local peak and global peak of RNase,
between the Gaussian curves of the main local peaks of Control and RNase.  

In addition, each protein was categorized as either *left*, *right* or *non-*shifting or *precipitated*. The shifting behavior was defined as following:
$\Delta\mu = \mu(Control)-\mu(RNase)> +1$: left shift; $\Delta\mu = \mu(Control)-\mu(RNase)< -1$: right shift; $-1 \leq\Delta\mu = \mu(Control)-\mu(RNase)\leq +1$: no shift; $\mu(Control)>24 \lor \mu(RNase)>24$: precipitated.  
  
### 2.5.2 Principal Component Analysis
Principal component analysis (PCA) was applied to the percentage values of the selection criteria mentioned above, so that they would be comparable to each other. The variables regarding main local peaks only served as determining partially shifting proteins and were therefore not included in PCA. The selection criteria which separated the proteins the most based on their shifting behavior in the plot of the first two principal components were chosen as optimal selection criteria for further analysis.  

### 2.5.3 Application of Selection Criteria  
Based on the PCA, four parameters were defined as most important for separating shifting from non-shifting proteins. The following thresholds for the four parameters were consecutively set for the classification of an RDP:

  1. Global peak shift of more than **1 fraction**
  2. Overlap of the Gaussian curves of less than **75 %**
  3. Protein loss under Control global peak of more than **20 %** after RNase treatment
  4. Protein gain under RNase global peak of more than **20 %** after RNase treatment
  
Furthermore, *partial* RDPs were defined as proteins that satisfy the first and fourth criterion and either the second or third criterion for RNA-dependence. More *partial* RDPs were identified using the following criteria: 

  1. Fraction shift between global and main local peak positions of more than **1 fraction**
  2. Overlap between the Gaussian curves of global and main local peaks of less than **75 %**
  3. Overlap between the Gaussian curves of the local peaks of less than **75 %**
  4. Protein gain under main local peak in relation to protein distribution of other sample under the same peak of more than **20 %**
  5. Protein amount at main local peak of more than **8**
  6. Previous categorization as **"non-RDP"**


## 2.6 Further Analysis

### 2.6.1 Comparison with the RBP2GO Database    
To evaluate the accuracy of the performed analysis, the identified RDPs were compared with their respective RBP-categorization in the RBP2GO database. 


### 2.6.2.1 Direct Comparison  
To further examine the relation between our RDP identification method and the RBP2GO classification, both RBP2GO-RBPs and -non-RBPs were plotted in the biplot of the first two principal components from PCA.


### 2.6.2.2 Cluster Comparison with RBP2GO Score and t-Test  
By calculating the **mean RBP2GO score** for each cluster (non-/partial/RDP) identified by applying the optimal selection criteria, the RNA-dependence of a cluster was estimated. Using a two-sided Welch t-test, the significance of the differences between the mean RBP2GO score values were calculated. The H0 and H1 hypotheses were defined as following: H0: The mean values are not significantly different from each other; H1: The mean values are significantly different from each other. The false positive rate was set as p = 0.05. 

### 2.6.2.3 Barplot with all Identified RDPs
Finally, a barplot was created to compare the overlap between all identified RDPs via the performed analysis and all RDPs classified as RBPs in the RBP2GO database.

### 2.6.3 Linear Regression    
In addition, it was investigated if the parameters calculated from mass spectrometry data could be used to examine the kinetics of RNA-protein interactions. To do this, *binding affinity* was defined as the difference in protein amount between the Control and RNase samples at the fraction position of the Control global peak and *binding specificity* was defined as the reciprocal value of the full width at half maximum of the Gaussian curve fitted to the Control global peak. Linear regression was used to calculate how well of a predictor binding affinity was to binding specificity.


# 3. Results

## 3.1 Results of Data Cleanup, Localization of Maxima and Gaussian Fitting
After performing all data cleanup steps, the original 7086 proteins were reduced to 5990 proteins. Local and global maxima were determined and Gaussian curves fitted to the peaks. As can be observed in the plots of the protein *"CTCF_HUMAN"*, the smoothing and normalization steps (*Fig. 1*) as well as the peak determination and Gaussian fitting (*Fig. 2*) were successful.

```{r smoothed and normalized, echo=FALSE, out.width = "33%"}


### not smoothed and normalized
Gene_Plotter("CTCF_HUMAN", Gauss =F, conf =F, Type= "Raw", grid=F, Gbl.Max=F, Loc.Max=F, Legend=F, mar=c(5,4,4,4))


### smoothed

Gene_Plotter("CTCF_HUMAN", Gauss=F, conf = F, Type="Smooth", grid=F, Gbl.Max=F, Loc.Max=F, Legend=F,mar=c(5,4,4,4))


### smoothed and normalized
Gene_Plotter("CTCF_HUMAN", Gauss=F, conf=F, Type="Smooth and Norm", grid=F, Gbl.Max=F, Loc.Max=F, Legend=T, Leg.Pos = c(0.05, 0),mar=c(5,4,4,7))

```
\footnotesize  
**Figure 1: Data Cleanup.** The protein distribution of the gene *CTCF_HUMAN* (*CCCTC-binding factor*) were smoothed and then normalized.\normalsize 
<br>  
<br>  

```{r maxima and Gaussian fit, echo=FALSE, out.width = "65%"}
### Gaussian fitted
Gene_Plotter("CTCF_HUMAN", grid =F, Leg.Pos = c(0.05, 0))

```
<br>  
\footnotesize
**Figure 2: Local and Global Maxima and Gaussian Fitting.** Local and global maxima of the protein *CCCTC-binding factor (CTCF_HUMAN)* together with fitted Gaussian curves. The grey area indicates the 95% confidence interval of the mean.
\normalsize 

## 3.2 Shifting Behavior
In *Fig. 3*, the shifting behavior is represented by plotting the global peak positions of the Control against those of the RNase samples. Non-shifting proteins lie in a straight line through the origin with a slope of 1, signifying equal positions of both global peaks. Left shifting proteins lie below this line, right shifting proteins lie above. Precipitated proteins lie close to the right as well as to the top edge which means that either one of the peaks is in the 25th fraction. Most proteins are non-shifting (5153 proteins). The largest group of shifting proteins are left shifting (782 proteins). Only a few proteins are right shifting (41 proteins) and even fewer are precipitated (14 proteins).

```{r shifting behavior, echo=FALSE, out.width = "65%"}

# peak positions und Einfärben nach "left", "right", "precipitated" und "no" shifting

col = rep(NA, nrow(Ctrl_RDeep))
col[which(Sel_Crit[, 25] == "left")] = "#6699FF"
col[which(Sel_Crit[, 25] == "right")] = "red"
col[which(Sel_Crit[, 25] == "precipitated")] = "orange"
col[which(Sel_Crit[, 25] == "no")] = "grey"

par(mar = c(4,5,5,9.5), xpd = T)

plot(Ctrl_Gauss_Fit[, 1], RNase_Gauss_Fit[, 1], pch = 16, col = col, main = "Shifting Behavior", xlab = "Control Global Peak Position", ylab = "RNase Global Peak Position")
legend("topright", inset = c(-0.47, 0), legend = c(paste("left (", length(which(Sel_Crit[, 25] == "left")), ")", sep =""), paste("right (", length(which(Sel_Crit[, 25] == "right")), ")", sep =""), paste("precipitated (", length(which(Sel_Crit[, 25] == "precipitated")), ")", sep =""), paste("no (", length(which(Sel_Crit[, 25] == "no")), ")", sep ="")), pch = 16, col = c("#6699FF", "red", "orange", "grey"))


```
<br>  
\footnotesize  
**Figure 3: Shifting Behavior.** All analyzed proteins were classified into left, right, and non-shifting, and precipitated using the criteria mentioned above. They were plotted using the global peak positions of the Control and RNase samples.\normalsize  

## 3.3 Results of PCA
In *Fig. 4 (left)* the PCA was applied to all selection criteria and the first two principle components were plotted against each other (also included in appendix as *Fig. A.1*). The different shifting behaviors of the proteins are shown by using different colors. Left shifting proteins are mostly separated from non-shifting proteins along the following four selection criteria: delta_Glb_mu_%, Ctrl_Gbl_prt_prop_%, RNase_Gbl_prt_prop_%, and negative Gbl_overlap_% (the point density is illustrated in *Fig. A.2*). Therefore, they were defined as the optimal selection criteria. To illustrate this, *Fig. 4 (right)* shows the first two principle components plotted against each other when PCA is performed solely for the optimal selection criteria. As expected, the proteins separate into two rough clusters along these criteria.

```{r Principal components, echo=FALSE, out.width="50%"}

# PCA applied to all precentage selection criteria

pca = prcomp(Sel_Crit[, c(2, 4, 6, 8, 10, 12, 13)])

library(ggplot2)
library(factoextra)

fviz_pca_biplot(pca, axes = c(1,2) , label = "var", col.var = "black", repel = T, col.ind = Sel_Crit[,25], pointshape =16, pointsize =1.5, palette = c("#6699FF", "gray", "orange", "red"), labelsize=5, title = "PCA (All Selection Criteria)")+ theme(axis.text=element_text(size=12), axis.title=element_text(size=15), plot.title=element_text(size=18, face="bold", hjust=0.5), legend.text=element_text(size=12), legend.title=element_blank(), legend.box.background = element_rect(color="black", size=0.4), legend.position ="none")

# PCA applied to following 4 selection criteria: delta_Gbl_mu_%**, **Ctrl_Gbl_prt_prop_%**, **RNase_Gbl_prt_prop_%**, and **Gbl_overlap_%

pca = prcomp(Sel_Crit[, c(2, 10, 12, 13)])
library(ggplot2)
library(factoextra)

fviz_pca_biplot(pca, axes = c(1,2), label = "var", col.var = "black", repel = T, col.ind = Sel_Crit[,25], pointshape=16, pointsize=1.5, labelsize=5, palette = c("#6699FF", "gray", "orange", "red"), title ="PCA (Optimal Selection Criteria)")+ theme(axis.text=element_text(size=12), axis.title=element_text(size=15), plot.title=element_text(size=18, face="bold", hjust=0.5), legend.text=element_text(size=12), legend.title=element_blank(), legend.box.background = element_rect(color="black", size=0.4))




```
\footnotesize  
**Figure 4: PCA Biplots.** Principle component analysis of all selection criteria (left). Principle component analysis of optimal selection criteria (right). The first two principle components are plotted against each other, the shift classification of the proteins is shown via different colors.\normalsize 

## 3.4 Application of Selection Criteria and Determination of RDPs
Application of the optimal selection criteria classifies 539 proteins as RDPs, 274 as partial RDPs, and 5177 as non-RDPs (individual proteins are illustrated in *Fig. A.5*). All proteins are plotted in a biplot of the first two principle components and colored according to their classification (*Fig. 5, left* and also *Fig. A.4, left*). Two main clusters are formed, one containing RDPs (blue) and one containing non-RDPs and partial RDPs (gray and orange) (see also *k*-means clustering in *Fig. A.3*).

```{r, echo=F}
#Include locals too & show how including these righteously expands the number of RBPs
#Example: HNRH3_HUMAN, Heterogeneous nuclear ribonucleoprotein H3, RBP2GO score of 84,8, partial shifter
#and SRSF9_HUMAN, Serine/arginine-rich splicing factor 9, RBP2GO score 71.0
```

## 3.5 Comparison with the RBP2GO Database
To determine the effectiveness of our method, the proteins classified as RDPs in our analysis were compared to the proteins classified as RBPs in the RBP2GO database. Firstly, both RBP2GO-RDPs and -non-RDPs were plotted in the biplot of the first two principal components used in our analysis. As depicted in *Fig. 5 (right)*, the RBP2GO-classified RBPs distribute homogeneously across the two clusters with no clear observable trend (see also *Fig. A.4, right*). 

```{r comparison with RBP2GO Database, echo =F, out.width="50%"}

#n für die 4 ausgewählten Selektionskriterien
pca = prcomp(Sel_Crit[, c(2, 10, 12, 13)])

col = rep(NA, nrow(Sel_Crit))
col[which(Sel_Crit[, 26] == "yes")] = "blue"
col[which(Sel_Crit[, 26] == "partial")] = "orange"
col[which(Sel_Crit[, 26] == "no")] = "grey"

plot(pca$x[, 1], pca$x[, 2], pch = 16, col = col, main = "Our RDPs and Non-RDPs", xlab = "PC1", ylab = "PC2")
legend("topright", legend = c(paste("RDP (", length(which(Sel_Crit[, 26] == "yes")), ")", sep =""), paste("partial RDP (", length(which(Sel_Crit[, 26] == "partial")), ")", sep =""), paste("non-RDP (", length(which(Sel_Crit[, 26] == "no")), ")", sep ="")), pch = 16, col = c("blue","orange", "grey"))

col = rep(NA, nrow(Sel_Crit))
col[which(rownames(Sel_Crit) %in% rownames(Table_HS_RBP))] = "blue"
col[which(rownames(Sel_Crit) %in% rownames(Table_HS_Non_RBP))] = "grey"

plot(pca$x[, 1], pca$x[, 2], pch = 16, col = col, main = "RBP2GO RBPs and Non-RBPs", xlab = "PC1", ylab = "PC2")
legend("topright", legend = c(paste("RBP (", length(which(rownames(Sel_Crit) %in% rownames(Table_HS_RBP))), ")", sep =""), paste("non-RBP (", length(which(rownames(Sel_Crit) %in% rownames(Table_HS_Non_RBP))), ")", sep ="")), pch = 16, col = c("blue", "grey"))

rm(pca, col)

```
\footnotesize  
**Figure 5: Identification of RDPs and Comparison to RBP2GO.** The first two principle components of the optimal selection criteria PCA are plotted against each other. The classification of the proteins (RDP, partial RDP, non-RDP) is shown by using different colors (left). The classification of the proteins (RBP, non-RBP) from the RBP2GO database is shown by using different colors (right).\normalsize

## 3.6 Cluster Comparison with RBP2GO Score and t-Test
The average RBP2GO score of each cluster was calculated. The average RBP2GO score of the RDPs is 21.5 with a standard error of 0.9, the average score of the partial RDPs is 13.6 with a standard error of 0.9 and the average score of the non-RDPs is 6.6 with a standard error of 0.1.  
A two-sided Welch t-test was used to evaluate the significance of the differences between the mean RBP2GO scores: p(RDP, non-RDP) = 8.5e-54, p(RDP, partial RDP) = 8.2e-10, p(partial RDP, non-RDP) = 2.0e-12. All clusters have significant p-values since each p-value is smaller than the False Positive Rate of p = 0.05.  

## 3.7 Bar Plot
Comparing the RDPs identified via our presented analysis with the RDPs classified as such in the RBP2GO database, 631 proteins were identified as RDPs in both analyses, 2946 proteins were listed as RDPs in the database but were not identified as RDPs in our performed analysis, and 182 proteins which we determined as RDPs are not classified as such on the RBP2GO database (*Fig. 6*).
```{r Bar plot RBPs, echo=F, out.width="50%"}
# ---> Barplot 1: Vergleich unserer RDPs mit RBPs von der RBP2GO Datenbank
par(mar=c(1, 3, 3, 8))
barplot(
  matrix(
  c(
    length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "no"), ]) %in% rownames(Table_HS_Non_RBP))) +
    length(which(rownames(Ctrl_RDeep[which(!(rownames(Ctrl_RDeep) %in% rownames(Table_HS_RBP)) & !(rownames(Ctrl_RDeep) %in% rownames(Table_HS_Non_RBP))), ]) %in% rownames(Sel_Crit[which(Sel_Crit[, 26] == "no"), ]))), 
    # RED: which protein are identified as non-RDPs in our analysis and in RBP2GO? + which proteins from our dataset are not included in RBP2GO?
  
    length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "no"), ]) %in% rownames(Table_HS_RBP))), 
    # ORANGE: which of our non-RDPs are in the RBP dataset from RBP2GO?
  
    length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "yes" | Sel_Crit[, 26] == "partial"), ]) %in% rownames(Table_HS_RBP))),
    # YELLOW: which of our RDPs or partial RDPs are also listed in the RBP2GO databank?
  
    length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "yes" | Sel_Crit[, 26] == "partial"), ]) %in% rownames(Table_HS_Non_RBP))) + 
    length(which(rownames(Ctrl_RDeep[which(!(rownames(Ctrl_RDeep) %in% rownames(Table_HS_RBP)) & !(rownames(Ctrl_RDeep) %in% rownames(Table_HS_Non_RBP))), ]) %in% rownames(Sel_Crit[which(Sel_Crit[, 26] == "yes" | Sel_Crit[, 26] == "partial"), ]))), 
    # GREEN: which of our RDPs are not listed in the RBP2GO Databank?
  
    rep(NA, 4)),
  nrow = 4),
  axes = F,
  col = c("red", "orange", "yellow", "green"), 
  main = "Comparison with RBP2GO"
)
legend("topright", 
       fill = c("green", "yellow", "orange", "red"),
       inset = c(0.01, 0),
       legend = c( 
        paste("New RDPs (", 
              length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "yes" | Sel_Crit[, 26] == "partial"), ]) %in% rownames(Table_HS_Non_RBP))) + 
              length(which(rownames(Ctrl_RDeep[which(!(rownames(Ctrl_RDeep) %in% rownames(Table_HS_RBP)) & !(rownames(Ctrl_RDeep) %in% rownames(Table_HS_Non_RBP))), ]) %in% rownames(Sel_Crit[which(Sel_Crit[, 26] == "yes" | Sel_Crit[, 26] == "partial"), ]))), 
              ")", sep = ""),
        
        paste("RDPs (", 
              length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "yes" | Sel_Crit[, 26] == "partial"), ]) %in% rownames(Table_HS_RBP))), 
              ")", sep = ""), 
        
        paste("RBP2GO-RDPs (", 
              length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "no"), ]) %in% rownames(Table_HS_RBP))), 
              ")", sep = ""),
        
        paste("non-RDPs (", 
              length(which(rownames(Sel_Crit[which(Sel_Crit[, 26] == "no"), ]) %in% rownames(Table_HS_Non_RBP))) + 
              length(which(rownames(Ctrl_RDeep[which(!(rownames(Ctrl_RDeep) %in% rownames(Table_HS_RBP)) & !(rownames(Ctrl_RDeep) %in% rownames(Table_HS_Non_RBP))), ]) %in% rownames(Sel_Crit[which(Sel_Crit[, 26] == "no"), ]))), 
              ")", sep = ""))
)
```
<br>  
\footnotesize  
**Figure 6: Bar Plot.** The new proteins classified as RDPs via the performed analysis and the RBP2GO database are illustrated in this bar plot.\normalsize

## 3.8 Linear Regression
Performing a linear regression to test whether binding affinity is a good predictor of binding specificity returns a very low coefficient of determination for all analyzed proteins (R² = 0.03) (*Fig. 7, left*) and a rather low coefficient of determination for only the RDPs (R² = 0.29) (*Fig. 7, right*). It is noticeable that the calculated correlation between binding affinity and specificity is positive in both cases, meaning that on average the reciprocal value of the full width at half maximum decreases with an increasing difference between the protein amount of Control and RNase samples at the global Control peak position.

```{r Linear Regression, echo=F, out.width="50%"}
RNase_h = sapply(seq(1, nrow(Sel_Crit), 1), function(n) {RNase_Mean_Smooth_Norm100[n, unlist(round(Ctrl_Mean_Gbl_Max))[n]]})

affinity = unlist(Ctrl_Gauss_Fit[, 3]) / (sqrt(2 * pi) * unlist(Ctrl_Gauss_Fit[, 2]))  - RNase_h
specifity = (2 * sqrt(2 * log(2)) * Ctrl_Gauss_Fit[, 2])^(-1)

fit <- lm(specifity ~ affinity)

col = rep(NA, length(Sel_Crit[, 26]))
col[which(Sel_Crit[, 26] == "yes")] = "blue"
col[which(Sel_Crit[, 26] == "partial")] = "orange"
col[which(Sel_Crit[, 26] == "no")] = "grey"
par(mar = c(5, 4, 4, 10.5), xpd = F)
plot(affinity, specifity, col = col, pch = 16, ylab = "Binding Specificity [1/FWHM]", xlab = "Binding Affinity [Normalized Protein Amount]")
curve(fit$coefficients[1] + fit$coefficients[2]*x, add = T, lwd = 3, col = "red")
par(xpd = T)
legend("topright", inset = c(-0.55, 0), legend = c("RDP", "partial RDP", "non-RDP", "Regression line", paste("y = ", round(fit$coefficients[2], 3), "x + ", round(fit$coefficients[1], 2), sep = ""), paste("R² = ", round(summary(fit)$r.squared, 2), sep = "")), pch = c(16, 16, 16, NA, NA, NA), col = c("blue", "orange", "grey", "red", NA, NA), lwd = c(NA, NA, NA, 3, NA, NA))

RNase_h = sapply(which(Sel_Crit[, 26] == "yes"), function(n) {RNase_Mean_Smooth_Norm100[n, unlist(round(Ctrl_Mean_Gbl_Max))[n]]})

affinity = Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "yes"), 3] / (sqrt(2 * pi) * Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "yes"), 2])  - RNase_h
specifity = (2 * sqrt(2 * log(2)) * Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "yes"), 2])^(-1)

fit <- lm(specifity ~ affinity)

col = rep(NA, length(which(Sel_Crit[, 26] == "yes")))
col[which(Sel_Crit[which(Sel_Crit[, 26] == "yes"), 26] == "yes")] = "blue"
col[which(Sel_Crit[which(Sel_Crit[, 26] == "yes"), 26] == "partial")] = "orange"
col[which(Sel_Crit[which(Sel_Crit[, 26] == "yes"), 26] == "no")] = "grey"
par(mar = c(5, 4, 4, 10.5), xpd = F)
plot(affinity, specifity, col = col, pch = 16, ylab = "Binding Specificity [1/FWHM]", xlab = "Binding Affinity [Normalized Protein Amount]")
curve(fit$coefficients[1] + fit$coefficients[2]*x, add = T, lwd = 3, col = "red")
par(xpd = T)
legend("topright", inset = c(-0.515, 0), legend = c("RDP", "Regression line", paste("y = ", round(fit$coefficients[2], 2), "x + ", round(fit$coefficients[1], 2), sep = ""), paste("R² = ", round(summary(fit)$r.squared, 2), sep = "")), pch = c(16, NA, NA, NA), col = c("blue", "red", NA, NA), lwd = c(NA, 3, NA, NA))

rm(affinity, RNase_h, fit, col, specifity)
```
\footnotesize  
**Figure 7: Linear Regression.** Linear regression was applied to the binding affinity and binding specificity for all proteins (left) and for only RDPs (right) classified in this analysis.\normalsize


# 4. Discussion
In conclusion, different parameters were calculated from the protein distributions provided by Caudron-Herger *et al.* for HeLa cells synchronized in interphase. Based on these parameters the RNA-dependence of these proteins was determined.  
One step in this analyis was smoothing (*Fig. 1*). Smoothing was performed to help with global and local peak determination because the averaged protein distributions were sometimes too jagged for unambiguous peak determination (*Fig. 2*). However, this approach has three problems: First, single fraction peaks result, upon smoothing, in multiple peaks in neighboring fractions. This problem was fixed by calculating the mean fraction position for multiple peaks. Second, smoothing sometimes caused the fraction position of peaks to change. But this did not have an influence on analysis, because the change in fraction position between raw and smoothed data was very slight (no more than one fraction position). Third, smoothing changed the protein amount which sometimes led to global peaks in the raw data becoming local peaks in the smoothed data and *vice versa*. This problem was partly fixed by defining additional selection criteria for partial RDPs using the main local peaks. 
  
Next, the shifting behavior of the proteins was analyzed. There were 782 left shifting, 41 right shifting, 5153 non-shifting, and 14 precipitated proteins (*Fig. 3*).  
  
PCA helped with determining the optimal selection criteria for an RDP, namely the fraction shift between global peaks, the overlap of the Gaussian curves fitted to the global peaks and the protein amount gained or lost under the global peaks after treatment with RNase (*Fig. 4*).  
  
Application of the selection criteria resulted in 539 RDPs, 274 partial RDPs, and 5177 non-RDPs (*Fig. 5, left*). 
These results compare well with the genes’ identified shifting behavior of which 823 are categorized as either right  or left shifting compared to 813 identified RDPs and partial RDPs. Note that this does not mean that the individual genes are classified in the same way (compare *Fig. 4, right* with *Fig. 5, left*). 

The identified RDPs are only partly comparable to the RBP2GO database (813 RDPs and partial RDPs and 5177 non-RDPs compared to 3577 RDPs and 2393 non-RDPs) (*Fig. 5*).   Another way to compare the identified RDPs with the RBP2GO database is the RBP2GO score. The mean of the RBP2GO scores for the identified RDPs was 21.5, compared to 13.6 for partial RDPs and 6.6 for non-RDPs. The differences in the mean values between RDPs, partial RDPs, and non-RDPs are all highly significant based on p-values calculated using a two-sided Welch t-test. It is noticeable that the p-value of the t-test for the difference between the mean values of the RBP2GO score of RDPs and non-RDPs is by far the smallest (p(RDP, non-RDP) = 8.5e-54). But these findings are of limited use because the RBP2GO score was calculated very differently from the analysis presented here (Caudron-Herger *et al.*, 2020a).  
  
The identified RDPs were also compared directly to the RBP2GO database (*Fig. 6*): 2231 genes (37 % of all analyzed genes) which are categorized as non-RDPs are also listed as non-RBPs in the RBP2GO database. In addition, 631 (11 %) RDPs are included as RBPs in RBP2GO. Yet, nearly half of the analyzed genes (2946 (49 %)) are categorized as non-RDPs, even though RBP2GO lists them as RBPs. Moreover, 182 (3 %) RDPs are listed as non-RBPs in the RBP2GO database. These "new" RDPs include proteins from which it is known that they are associated with RNA (e.g. *CDK13_HUMAN*, *Fig. A.6*) (Berro *et al.*, 2008).  

The differences between these results and the RBP2GO database are possibly the result of different human cell lineages used, different stages of the cell cycle analyzed and different experimental methods. Notably, only the proteome present in interphase was analyzed here, so poor comparability with the RBP2GO database which draws its data from many orthogonal experiments should be expected.   

Lastly, it was tested whether kinetic parameters could be determined using the mass spectrometry data. It would be expected that an RDP which binds tightly to its RNA target also forms very specific structures and, therefore, has a high binding specificity. This was partially observed when applying linear regression modeling only to the identified RDPs (*Fig. 7, right*). Analysis of all proteins or partial RDPs or non-RDPs show a lower coefficient of correlation (*Fig. 7, left* and *Fig. A.7*). However, the experiment was not set up to investigate the kinetics of RNA-protein interactions; therefore, these findings are of no particular use. Other methods have been developed to investigate the binding kinetics between RNA and proteins (Sharma *et al.*, 2021).  

The results of this analysis show that, besides the experimental methods used, bioinformatic analysis also plays a major role in determining the RNA-dependence of proteins. Further improvements in analysis of RNA-protein interactions will undoubtedly lead to better understanding and new therapies of human diseases associated with RBPs and RDPs. 
\newpage

# 5. References
Berro, R., Pedati, C., Kehn-Hall, K., Wu, W., Klase, Z., Even, Y., Genevière, A.M., Ammosova, T., Nekhai, S., and Kashanchi, F. (2008). CDK13, a new potential human immunodeficiency virus type 1 inhibitory factor regulating viral mRNA splicing. J. Virol. *82*, 7155-716.

Burd, C.G., and Dreyfuss, G. (1994). Conserved Structures and Diversity of Functions of RNA-Binding Proteins. Science *265*, 615-621.

Caudron-Herger, M., Rusin, S.F., Adamo, M.E., Seiler, J., Schmid, V.K., Barreau, E., Kettenbach, A.N., and Diederichs, S. (2019). R-DeeP: Proteome-wide and Quantitative Identification of RNA-Dependent Proteins by Density Gradient Ultracentrifugation. Mol. Cell *75*, 184-199.

Caudron-Herger, M., Jansen, R.E., Wassmer, E., and Diederichs, S. (2020a). RBP2GO: a comprehensive pan-species database on RNA-binding proteins, their interactions and functions. Nucleic Acids Res. *49*, D425-D436.

Caudron-Herger, M., Wassmer, E., Nasa, I., Schultz, A.-S., Seiler, J., Kettenbach, A.N., and Diederichs, S. (2020b). Identification, quantification and bioinformatic analysis of RNA-dependent proteins by RNase treatment and density gradient ultracentrifugation using R-DeeP. Nat. Protoc. *15*, 1338-1370.

Gebauer, F., Schwarzl, T., Valcárcel, J., and Hentze, M.W. (2021). RNA-binding proteins in human genetic disease. Nat. Rev. Genet. *22*, 185-198.

Gerstberger, S., Hafner, M., and Tuschl, T. (2013). Learning the language of post-transcriptional gene regulation. Genome Biol. *14*, 130.

Gerstberger, S., Hafner, M., and Tuschl, T. (2014). A census of human RNA-binding proteins. Nat. Rev. Genet. *15*, 829-845.

Jankowsky, E., and Harris, M.E. (2015). Specificity and nonspecificity in RNA–protein interactions. Nat. Rev. Mol. Cell Biol. *16*, 533-544.

Licatalosi, D.D., Ye, X., and Jankowsky, E. (2020). Approaches for measuring the dynamics of RNA–protein interactions. WIREs RNA *11*, e1565.

Lunde, B.M., Moore, C., and Varani, G. (2007). RNA-binding proteins: modular design for efficient function. Nat. Rev. Mol. Cell Biol. *8*, 479-490.

Sharma, D., Zagore, L.L., Brister, M.M., Ye, X., Crespo-Hernández, C.E., Licatalosi, D.D., and Jankowsky, E. (2021). The kinetic landscape of an RNA-binding protein in cells. Nature *591*, 152-156.
\newpage

# 6. Appendix

```{r Appendix Shifting Behavior, echo=FALSE, out.width="50%"}
# shifiting behavior PCA all sel. criteria
pca = prcomp(Sel_Crit[, c(2, 4, 6, 8, 10, 12, 13)])

# Plot, der die Datenpunkte je nach Kategorisierung in "left", "right" und "no" shift einfärbt 
col = rep(NA, nrow(Sel_Crit))

col[which(Sel_Crit[, 25] == "left")] = "#6699FF"
col[which(Sel_Crit[, 25] == "right")] = "red"
col[which(Sel_Crit[, 25] == "precipitated")] = "orange"
col[which(Sel_Crit[, 25] == "no")] = "grey"


plot(pca$x[, 1], pca$x[, 2], pch = 16, col = col, main = "Shifting Behavior with all Selection Criteria", xlab = "PC1", ylab = "PC2")
legend("bottomright", legend = c(paste("left (", length(which(Sel_Crit[, 25] == "left")), ")", sep =""), paste("right (", length(which(Sel_Crit[, 25] == "right")), ")", sep =""), paste("precipitated (", length(which(Sel_Crit[, 25] == "precipitated")), ")", sep =""), paste("no (", length(which(Sel_Crit[, 25] == "no")), ")", sep ="")), pch = 16, col = c("#6699FF", "red", "orange", "grey"))



# shifiting behavior 4 sel. criteria

pca = prcomp(Sel_Crit[, c(2, 10, 12, 13)])

#  Plot, der die Datenpunkte je nach Kategorisierung in "left", "right" und "no" shift einfärbt 
col = rep(NA, nrow(Sel_Crit))
col[which(Sel_Crit[, 25] == "left")] = "#6699FF"
col[which(Sel_Crit[, 25] == "right")] = "red"
col[which(Sel_Crit[, 25] == "precipitated")] = "orange"
col[which(Sel_Crit[, 25] == "no")] = "grey"

par(mar = c(4,5,5,9.5), xpd = T)
plot(pca$x[, 1], pca$x[, 2], pch = 16, col = col, main = "Shifting Behavior with Optimal Selection Criteria", xlab = "PC1", ylab = "PC2")

legend("topright", legend = c(paste("left (", length(which(Sel_Crit[, 25] == "left")), ")", sep =""), paste("right (", length(which(Sel_Crit[, 25] == "right")), ")", sep =""), paste("precipitated (", length(which(Sel_Crit[, 25] == "precipitated")), ")", sep =""), paste("no (", length(which(Sel_Crit[, 25] == "no")), ")", sep ="")), pch = 16, col = c("#6699FF", "red", "orange", "grey"))

```
\footnotesize  
**Figure A.1: PCA with Shifting Behavior.** The first two principle components of the PCA applied to all selection criteria (left) and of the PCA applied to the optimal selection criteria (right) are plotted against each other. The shifting classification of the proteins is shown by different colors.\normalsize

<br>  
<br>  

```{r Appendix Point Density, echo=FALSE, out.width="50%"}

# shifiting behavior PCA all sel. criteria
pca = prcomp(Sel_Crit[, c(2, 4, 6, 8, 10, 12, 13)])

#  Plot, der die Dichte der Punkteverteilung im Plot durch Farben anzeigt ("rot" = sehr dicht beieinander; "blau" = nicht dicht beieinander)

par(mar = c(5, 4, 4, 7), xpd = F)
plot(pca$x[, 1], pca$x[, 2], pch = 16, col = densCols(pca$x[, 1], pca$x[, 2], colramp = colorRampPalette(c("blue", "cyan", "green", "yellow", "orange", "red"))), main = "Point Density of All Selection Criteria", xlab = "PC1", ylab = "PC2")
abline(h = 0, lty =3, col = "grey")
abline(v = 0, lty = 3, col = "grey")

lgd_ = rep(NA, 100)
lgd_[c(1,100)] = c("Low","High")
par(xpd = T)
legend("topright",
       inset = c(-0.22,0),
       legend = lgd_,
       fill = colorRampPalette(c("blue", "cyan", "green", "yellow", "orange", "red"))(100),
       border = NA,
       y.intersp = 0.05,
       box.col = NA)

# shifitng behavior 4 sel. criteria
pca = prcomp(Sel_Crit[, c(2, 10, 12, 13)])

#  Plot, der die Dichte der Punkteverteilung im Plot durch Farben anzeigt ("rot" = sehr dicht beieinander; "blau" = nicht dicht beieinander)
par(mar = c(5, 4, 4, 7), xpd = F)
plot(pca$x[, 1], pca$x[, 2], pch = 16, col = densCols(pca$x[, 1], pca$x[, 2], colramp = colorRampPalette(c("blue", "cyan", "green", "yellow", "orange", "red"))), main = "Point Density of Optimal Selection Criteria", xlab = "PC1", ylab = "PC2")
abline(h = 0, lty =3, col = "grey")
abline(v = 0, lty = 3, col = "grey")

lgd_ = rep(NA, 100)
lgd_[c(1,100)] = c("Low","High")
par(xpd = T)
legend("topright",
       inset = c(-0.22,0),
       legend = lgd_,
       fill = colorRampPalette(c("blue", "cyan", "green", "yellow", "orange", "red"))(100),
       border = NA,
       y.intersp = 0.05,
       box.col = NA)
```
\footnotesize  
**Figure A.2: PCA Densitiy Plots.** The first two principle components of the PCA applied to all selection criteria (left) and of the PCA applied to the optimal selection criteria (right) are plotted against each other. The point density of the proteins is shown by the usage of different colors.\normalsize

<br>  
<br>  


```{r k-Means Clustering, echo=FALSE, out.width="50%"}
pca = prcomp(Sel_Crit[, c(2, 10, 12, 13)])

kmeans = kmeans(pca$x, 2, iter.max = 10, nstart = 1)

col = rep(NA, nrow(Sel_Crit))
col[which(kmeans$cluster == 1)] = "blue"
col[which(kmeans$cluster == 2)] = "red"

par(mar = c(5, 4, 4, 2) + 0.1)
plot(pca$x[, 1], pca$x[, 2], pch = 16, col = col, main = "k-Means Clustering (n = 2)", xlab = "PC1", ylab = "PC2")
par(xpd=T)
legend("topright", 
       legend = c(paste("Cluster 1 (", length(which(kmeans$cluster == 1)), ")", sep = ""), paste("Cluster 2 (", length(which(kmeans$cluster == 2)), ")", sep = "")), 
       pch = 16, 
       col = c("blue", "red"))

# ---> Erstellung eines Elbow-
par(mar = c(5, 4, 4, 2) + 0.1)
plot(1:5, c(kmeans(pca$x[,1:2], 1, iter.max = 10, nstart = 1)$tot.withinss,
            kmeans(pca$x[,1:2], 2, iter.max = 10, nstart = 1)$tot.withinss,
            kmeans(pca$x[,1:2], 3, iter.max = 10, nstart = 1)$tot.withinss,
            kmeans(pca$x[,1:2], 4, iter.max = 10, nstart = 1)$tot.withinss,
            kmeans(pca$x[,1:2], 5, iter.max = 10, nstart = 1)$tot.withinss), type = "b", pch = 19, main = "Elbow Method", xlab = "Number of cluster", ylab = "Total within-cluster sum of squares")
```
\footnotesize  
**Figure A.3:** ***k*****-Means Clustering of PCA Biplot of Optimal Selection Criteria.** The elbow plot for determination of optimal number of clusters (right) and applied clusters to the biplot of the optimal selection criteria from PCA (left).\normalsize

<br>  
<br>  

```{r comparison with RBP2GO Database appendix, echo =F, out.width="50%"}
pca = prcomp(Sel_Crit[, c(2, 4, 6, 8, 10, 12, 13)])

# ---> Erstellen eines Farbenvektors basierend auf den Kategorien "yes", "partial" und "no" (bezogen auf RDP)
col = rep(NA, nrow(Sel_Crit))
col[which(Sel_Crit[, 26] == "yes")] = "blue"
col[which(Sel_Crit[, 26] == "partial")] = "orange"
col[which(Sel_Crit[, 26] == "no")] = "grey"

# ---> Generieren des PCA biplots mit entsprechender Einfärbung
plot(pca$x[, 1], pca$x[, 2], pch = 16, col = col, main = "Our RDPs and Non-RDPs", xlab = "PC1", ylab = "PC2")
legend("bottomright", legend = c(paste("RDP (", length(which(Sel_Crit[, 26] == "yes")), ")", sep =""), paste("partial RDP (", length(which(Sel_Crit[, 26] == "partial")), ")", sep =""), paste("non-RDP (", length(which(Sel_Crit[, 26] == "no")), ")", sep ="")), pch = 16, col = c("blue","orange", "grey"))

# ---> Vergleich mit RBP2GO Datenbank
col = rep(NA, nrow(Sel_Crit))
col[which(rownames(Sel_Crit) %in% rownames(Table_HS_RBP))] = "blue"
col[which(rownames(Sel_Crit) %in% rownames(Table_HS_Non_RBP))] = "grey"

plot(pca$x[, 1], pca$x[, 2], pch = 16, col = col, main = "RBP2GO RBPs and Non-RBPs", xlab = "PC1", ylab = "PC2")
legend("bottomright", legend = c(paste("RBP (", length(which(rownames(Sel_Crit) %in% rownames(Table_HS_RBP))), ")", sep =""), paste("non-RBP (", length(which(rownames(Sel_Crit) %in% rownames(Table_HS_Non_RBP))), ")", sep ="")), pch = 16, col = c("blue", "grey"))
```
\footnotesize  
**Figure A.4: Identification of RDPs and Comparison to RBP2GO.** The first two principle components of the PCA for all selection criteria are plotted. The classification of the proteins (RDP, partial RDP, non-RDP) achieved in the performed analysis is shown by using different colors (left). The classification of the proteins (RDP, non-RDP) from the RBP2GO database is shown by using different colors (right).\normalsize
\newpage  


```{r Exemplary Graphs, echo=FALSE, out.width="50%"}
#non-RDP:
Gene_Plotter("PZP_HUMAN", cus.main = "PZP_HUMAN (non-RDP)", Leg.Pos = c(0.05, 0), grid = F)

# RDP:
Gene_Plotter("HNRPL_HUMAN", cus.main = "HNRPL_HUMAN (RDP)", Leg.Pos = c(0.05, 0), grid = F)

# partial RDP: 
# categorized based on local peaks
Gene_Plotter("HEXI1_HUMAN", cus.main = "HEXI1_HUMAN (partial RDP)", Leg.Pos = c(0.05, 0), grid = F)
# categrozied based on 2. | 3.
Gene_Plotter("RTCA_HUMAN", cus.main = "RTCA_HUMAN (partial RDP)", Leg.Pos = c(0.05, 0), grid = F)
```
\footnotesize  
**Figure A.5: Application of Selection Criteria.** Protein and Gaussian curves for the Control and RNase sample for the following four different proteins are plotted for comparison: *Pregnancy zone protein (PZP_HUMAN)* (non-RDP), *Heterogeneous nuclear ribonucleoprotein L (HNRPL_HUMAN)* (RDP), *Hexamethylene bisacetamide-inducible protein 1 (HEXI1_HUMAN)* (partial RDP), *RNA 3'-terminal phosphate cyclase (RTCA_HUMAN)* (partial RDP).\normalsize

<br>  
<br>  

```{r New RDP with RNA function, echo=FALSE, out.width="50%"}
Gene_Plotter("CDK13_HUMAN", grid = F, Leg.Pos = c(0.05, 0))
# CDK13_HUMAN: Cyclin-dependent kinase 13. "Cyclin-dependent kinase which displays CTD kinase activity and is required for RNA splicing"
```
<br>  

\footnotesize  
**Figure A.6: New RDP with RNA Associating.** *Cyclin-dependent kinase 13 (CDK13_HUMAN)* is involved in many key cellular RNA processes such as splicing and transcription (Berro *et al.*, 2008).\normalsize

<br>  
<br>  

```{r Linear Regression Appendix, echo=F, out.width="50%"}
RNase_h = sapply(which(Sel_Crit[, 26] == "partial"), function(n) {RNase_Mean_Smooth_Norm100[n, unlist(round(Ctrl_Mean_Gbl_Max))[n]]})

affinity = Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "partial"), 3] / (sqrt(2 * pi) * Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "partial"), 2])  - RNase_h
specifity = (2 * sqrt(2 * log(2)) * Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "partial"), 2])^(-1)

fit <- lm(specifity ~ affinity)

col = rep(NA, length(which(Sel_Crit[, 26] == "partial")))
col[which(Sel_Crit[which(Sel_Crit[, 26] == "partial"), 26] == "partial")] = "blue"
col[which(Sel_Crit[which(Sel_Crit[, 26] == "partial"), 26] == "partial")] = "orange"
col[which(Sel_Crit[which(Sel_Crit[, 26] == "partial"), 26] == "no")] = "grey"
par(mar = c(5, 4, 4, 10.5), xpd = F)
plot(affinity, specifity, col = col, pch = 16, ylab = "Binding Specificity [1/FWHM]", xlab = "Binding Affinity [Normalized Protein Amount]")
curve(fit$coefficients[1] + fit$coefficients[2]*x, add = T, lwd = 3, col = "red")
par(xpd = T)
legend("topright", inset = c(-0.54, 0), legend = c("partial RDP", "Regression line", paste("y = ", round(fit$coefficients[2], 4), "x + ", round(fit$coefficients[1], 2), sep = ""), paste("R² = ", round(summary(fit)$r.squared, 3), sep = "")), pch = c(16, NA, NA, NA), col = c("orange", "red", NA, NA), lwd = c(NA, 3, NA, NA))

RNase_h = sapply(which(Sel_Crit[, 26] == "no"), function(n) {RNase_Mean_Smooth_Norm100[n, unlist(round(Ctrl_Mean_Gbl_Max))[n]]})

affinity = Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "no"), 3] / (sqrt(2 * pi) * Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "no"), 2])  - RNase_h
specifity = (2 * sqrt(2 * log(2)) * Ctrl_Gauss_Fit[which(Sel_Crit[, 26] == "no"), 2])^(-1)

fit <- lm(specifity ~ affinity)

col = rep(NA, length(which(Sel_Crit[, 26] == "no")))
col[which(Sel_Crit[which(Sel_Crit[, 26] == "no"), 26] == "no")] = "blue"
col[which(Sel_Crit[which(Sel_Crit[, 26] == "no"), 26] == "partial")] = "orange"
col[which(Sel_Crit[which(Sel_Crit[, 26] == "no"), 26] == "no")] = "grey"
par(mar = c(5, 4, 4, 10.5), xpd = F)
plot(affinity, specifity, col = col, pch = 16, ylab = "Binding Specificity [1/FWHM]", xlab = "Binding Affinity [Normalized Protein Amount]")
curve(fit$coefficients[1] + fit$coefficients[2]*x, add = T, lwd = 3, col = "red")
par(xpd = T)
legend("topright", inset = c(-0.54, 0), legend = c("non-RDP", "Regression line", paste("y = ", round(fit$coefficients[2], 3), "x + ", round(fit$coefficients[1], 2), sep = ""), paste("R² = ", round(summary(fit)$r.squared, 2), sep = "")), pch = c(16, NA, NA, NA), col = c("grey", "red", NA, NA), lwd = c(NA, 3, NA, NA))

rm(affinity, RNase_h, fit, col, specifity)
```
\footnotesize  
**Figure A.7: Linear Regression for Partial RDPs and Non-RDPs.** Linear regression was applied to the binding affinity and the binding specificity for only partial RDPs (left) and for only non-RDPs (right). Both linear regressions have considerably smaller R² values compared to the linear regression of only RDPs.\normalsize
